doc: added note about brew and Qt 6 on macOS (#25948)
bitcoin/bitcoin: doc/build-osx.md

brew install qt@5
```

> _Note:_ Ensure that _only_ `qt@5` is installed:

Reviewer: **nit**: This seems to be the only place where "Note" is italicized in the doc; it might be worth reverting to match the other notes. Italicizing also seems redundant here for emphasis, as it's already a note. 

```suggestion
Note: Ensure that _only_ `qt@5` is installed:
```

Author: Done :)

> > If the `qt` package, which provides Qt 6, is installed alongside `qt@5`, configuring will find Qt 5, but compiling will fail because some headers are incorrectly imported from Qt 6 locations.

Reviewer 2: Please unquote this
```suggestion
If the `qt` package, which provides Qt 6, is installed alongside `qt@5`, configuring will find Qt 5, but compiling will fail because some headers are incorrectly imported from Qt 6 locations.
```

Reviewer 2: It's sufficient to say that you can't build if qt6 is installed, we don't need to go into any technical detail. The only information needed is that `qt@5` is the proper package and not `qt6`. I'd still prefer something like this:
```suggestion
If you've previously grabbed the `qt` package, which installs Qt 6, you will not be able to compile.
```

Author: @jarolrod I have removed the quote formatting as suggested.

Can you please explain why there is no need to go into rationale that you cannot have both Qt 5 and Qt 6 installed simultaneously?  After-all Qt 6 is becoming quite common now, and it quite reasonable that some developers will think: "Hey, why not? I need Qt 6 for this other project I'm working on.".

> You may check for and uninstall packages with the following commands:

Reviewer 2: ```suggestion
You may check for and uninstall the `qt` package with the following commands:
```

brew install qt@5
```

> Note: Ensure that _only_ `qt@5` is installed:
> 

Reviewer 3: nit: `brew list --version | grep qt` is the command to ensure that only qt@5 is installed, so maybe we could move it just after the CTA?
```suggestion
Note: Ensure that _only_ `qt@5` is installed:
``` bash
brew list --version | grep qt
```
```
